File: python/p003.py
"""
    
Project Euler - Problem 3 - Largest prime factor
https://projecteuler.net/problem=3

"""
import timing
import eulerutil


def main():
    n = 600851475143
    factors = eulerutil.prime_decomposition(n)
    print('%s' %factors[-1])
    

if __name__ == '__main__':
    main()

File: js/p003.js
"use strict";
const primeFactors = require('./util/tools.js').primeFactorization;

/*
 * Largest prime factor
 * Problem 3
 * The prime factors of 13195 are 5, 7, 13 and 29.
 * 
 * What is the largest prime factor of the number 600851475143 ?
 * 
 * Answer: 6857
 */

function init() {
	var input = 600851475143;
	var factors = primeFactors(input);

    return Object.keys(factors).sort(function(a,b) { return b-a;})[0];
}

module.exports.init = init;